src/test/java/com/puppycrawl/tools/checkstyle/internal/AllTestsTest.java

```diff
@@ -49,11 +49,11 @@ public void testAllInputsHaveTest() throws Exception {
 
         Assert.assertTrue("found tests", !allTests.keySet().isEmpty());
 
-        Files.walk(Paths.get("src/test/resources"))
+        Files.walk(Paths.get("src/test/resources/com/puppycrawl/tools/checkstyle"))
             .forEach(filePath -> {
                 verifyInputFile(allTests, filePath.toFile());
             });
-        Files.walk(Paths.get("src/test/resources-noncompilable"))
+        Files.walk(Paths.get("src/test/resources-noncompilable/com/puppycrawl/tools/checkstyle"))
```

Comment:
These lines were changed just to ignore `configuration.properties`.

I would rather see us continue to validate all folders and in this case ignore this specific file.
Please add location and reason to 
https://github.com/Softus/checkstyle/blob/e9f358c9d72e9898753dbca0c1cf6b865ce26158/src/test/java/com/puppycrawl/tools/checkstyle/internal/AllTestsTest.java#L251-L258
and revert these lines to what they were before.

Reply:
Currently, the test `AllTestTest` can not validate all folders. The method `verifyInputFile` starts with
```
    private static void verifyInputFile(Map<String, List<String>> allTests, File file) {
        if (file.isFile()) {
            final String path;

            try {
                path = getSimplePath(file.getCanonicalPath());
            }
            catch (IOException ex) {
                throw new IllegalStateException(ex);
            }
```
and `getSimplePath` is:
```
    private static String getSimplePath(String path) {
        return path.substring(path.lastIndexOf("com" + File.separator + "puppycrawl"));
    }
```

It will throw `IndexOutOfRangeException` for every path not starting with "com/puppycrawl".